tests/helpers/test_delete_instance.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+import pytest_rpc.helpers
+import pytest
+import json
+import testinfra.backend.base
+import testinfra.host
+
+"""Test cases for the 'delete_instance' helper function."""
+
+
+def test_successful_deletion(mocker):
+    """Verify delete_instance returns None when server instance is
+    successfully deleted."""
+
+    fake_backend = mocker.Mock(spec=testinfra.backend.base.BaseBackend)
+    myhost = testinfra.host.Host(fake_backend)
+    cr1 = mocker.Mock(spec=testinfra.backend.base.CommandResult)
+    cr2 = mocker.Mock(spec=testinfra.backend.base.CommandResult)
+    mocker.patch('testinfra.host.Host.run', side_effect=[cr1, cr2, cr2])
```

Comment:
I'm not sure I understand where `[cr1, cr2, cr2]` are being used. I see one of them ( the return from .run())

Reply:
> If side_effect is an iterable then each call to the mock will return the next value from the iterable.
> - Python docs

OK, I think I just answered my own question. other methods in this helper will call `.run()`

Reply:
Neat trick!

Reply:
The first call to `run` is triggered by the [get_id_by_name](https://github.com/rcbops/pytest-rpc/blob/master/pytest_rpc/helpers.py#L233) call and should return a CommandResult that includes the json with the server name and id. The second (and subsequent) call to `run` is triggered by the [asset_not_in_list](https://github.com/rcbops/pytest-rpc/blob/master/pytest_rpc/helpers.py#L238) call and should return CommandResult that does not include the server.